app/models/metric/ci_mixin/capture.rb

```diff
@@ -17,6 +17,20 @@ def queue_name_for_metrics_collection
     ems.metrics_collector_queue_name
   end
 
+  def split_capture_intervals(start_time, end_time)
+    intervals = []
+    start_hour = start_time
+    while start_hour != end_time
```

Comment:
Is there any validation on the end_time?  Looks like this could loop infinitely if start_time and end_time aren't aligned